src/graph/InsertEdgeExecutor.cpp

```diff
@@ -125,6 +137,78 @@ StatusOr<std::vector<storage::cpp2::Edge>> InsertEdgeExecutor::prepareEdges() {
+                                           << defaultValues_[fieldName];
+                                return Status::Error("Type Conversion Failed");
+                            }
+
+                            VLOG(3) << "Insert " << fieldName << ":" << variantType
+                                    << " at " << insertPosition;
+                            values.insert(values.begin() + insertPosition, std::move(variantType));
+                            break;
+                        case nebula::cpp2::SupportedType::INT:
+                            try {
+                                variantType = folly::to<int64_t>(defaultValues_[fieldName]);
+                            } catch (const std::exception& ex) {
+                                LOG(ERROR) << "Conversion to int64_t failed: "
+                                           << defaultValues_[fieldName];
+                                return Status::Error("Type Conversion Failed");
+                            }
+
+                            VLOG(3) << "Insert " << fieldName << ":" << variantType
+                                    << " at " << insertPosition;
+                            values.insert(values.begin() + insertPosition, std::move(variantType));
+                            break;
+                        case nebula::cpp2::SupportedType::DOUBLE:
+                            try {
+                                variantType = folly::to<double>(defaultValues_[fieldName]);
+                            } catch (const std::exception& ex) {
+                                LOG(ERROR) << "Conversion to double failed: "
+                                           << defaultValues_[fieldName];
+                                return Status::Error("Type Conversion Failed");
+                            }
+
+                            VLOG(3) << "Insert " << fieldName << ":" << variantType
+                                    << " at " << insertPosition;
+                            values.insert(values.begin() + insertPosition, std::move(variantType));
+                            break;
+                        case nebula::cpp2::SupportedType::STRING:
+                            variantType = defaultValues_[fieldName];
+
+                            VLOG(3) << "Insert " << fieldName << ":" << variantType
+                                    << " at " << insertPosition;
+                            values.insert(values.begin() + insertPosition, std::move(variantType));
+                            break;
+                        default:
+                            LOG(ERROR) << "Unknow type";
+                            return Status::Error("Unknow type");
+                    }
+                    insertPosition++;
+                } else if (*props_[propsIndex] == fieldName) {
+                    propsIndex++;
+                    insertPosition++;
+                } else {
+                    insertPosition++;
```

Comment:
Why we can increase the position here? It seems you handle the situation when `props_[propsIndex]` match with `fieldName` in if statement, and mismatch in previous else if. Forgive me, a little bit hard to comprehend.